### PR TITLE
separate code into src and tests directories

### PR DESCRIPTION
**Pull request by harryBedfordSL**
## Description

This PR separates out our test code from our source code. By doing this our docker file is slightly more optimised, as we are now only copying over the functional code to be part of the container; the tests are not needed.

## Changelog
- create new root level `src` directory
- create new root level `tests` directory
- Move all backend code into respective places
- Update the backend docker file

## Notes
- Running the backend tests remains unchanged: `pytest`
- I have verified that `docker compose up` is still working
- Now, to run the backend directly (from `/backend`) the uvicorn command changes to: `uvicorn src.api:app --port 8250`

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -69,13 +69,16 @@ To add further rules, these are added to `ruff.toml` by using the letter asssign
```
Comment (Ch0live) [inline backend/README.md:69]: Line 25 needs updating to
```
uvicorn src.api:app --port 8250
```

Comment (harryBedfordSL) [reply backend/README.md:69]: great spot!

Comment (harryBedfordSL) [reply backend/README.md:69]: updated 🙂

**Comment by Ch0live** [comment]
- Local run works
- Docker run works
- All (33) tests pass

Great stuff!

File: backend/src/utils/__init__.py [hunk 2]
```diff
@@ -1,12 +1,13 @@
 from .config import Config
-from .llm import call_model, call_model_with_tools, get_response_three_prompts
+from .llm import call_model, call_model_with_tools, get_response_three_prompts, MistralClient
```
Comment (Ch0live) [inline backend/src/utils/__init__.py:2]: Do we actually use `MistralClient` outside of the `/utils` directory?

**Review by Ch0live — CHANGES_REQUESTED** [review]
1 other comment about MistralClient added

File: backend/src/utils/__init__.py (hunk 2 above)
Comment (harryBedfordSL) [reply backend/src/utils/__init__.py:2]: we use it in the tests

**Review by Ch0live — DISMISSED** [review]

**Comment by bheyman-scottlogic** [comment]
I think that under the `/src` directory it should only be directories for ease of navigation and that shouldn't necessarily be a package with an `__init__.py` . Happy to discuss what you think is more appropriate?

**Review by Ch0live — APPROVED** [review]

**Review by bheyman-scottlogic — APPROVED** [review]
Looks good, nice work Harry!

**Pull request merged**